react/components/molecules/search/search.js

```diff
@@ -0,0 +1,143 @@
+            placeholder: "Search",
+            buttonProps: {},
+            onValue: () => {},
+            onFocus: () => {},
+            onBlur: () => {},
+            onClear: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: this.props.value,
+            buttonsVisible: true,
+            focused: false
+        };
+    }
+
+    focus = () => {
+        this.textInputComponent.focus();
+    };
+
+    blur = () => {
+        this.textInputComponent.blur();
+    };
+
+    clear = () => {
+        this.setState({ valueData: "" }, () => {
+            this.props.onValue("");
+            this.props.onClear();
+            this.focus();
+        });
+    };
+
+    _icon = () => {
+        return this.state.valueData ? "close" : "search";
+    };
+
+    onInputValue = value => {
+        this.setState({ valueData: value }, () => {
+            this.props.onValue(value);
+        });
+    };
+
+    onInputFocus = () => {
+        this.setState({ focused: true }, () => {
+            this.props.onFocus();
+        });
+    };
+
+    onInputBlur = () => {
+        this.setState({ focused: false }, () => {
+            this.props.onBlur();
+        });
+    };
+
+    onButtonPress = () => {
+        if (!this.state.valueData) return;
+        this.clear();
+    };
+
+    _style = () => {
+        return [styles.search, this.props.style];
+    };
+
+    render() {
+        return (
+            <View style={this._style()}>
+                <Input
+                    ref={el => (this.textInputComponent = el)}
+                    color={"#223645"}
+                    style={styles.textArea}
+                    value={this.state.valueData}
+                    placeholder={this.props.placeholder}
+                    placeholderTextColor={"#223645"}
+                    onValueUpdate={this.onInputValue}
+                    onFocus={this.onInputFocus}
+                    onBlur={this.onInputBlur}
+                />
+                <ButtonIcon
+                    style={styles.button}
+                    icon={this._icon()}
+                    size={24}
+                    iconStrokeColor={"#1d2631"}
+                    iconStrokeWidth={2}
+                    iconHeight={24}
+                    iconWidth={24}
+                    {...this.props.buttonProps}
+                    onPress={this.onButtonPress}
+                />
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    search: {
+        flexDirection: "row",
+        alignItems: "center",
+        position: "relative"
+    },
+    textArea: {
+        flex: 1,
+        paddingBottom: 0,
+        paddingTop: 0,
+        borderRadius: 6,
+        minHeight: 40,
+        backgroundColor: "#ffffff"
+    },
+    button: {
+        position: "absolute",
+        backgroundColor: "transparent",
+        top: Platform.OS === "ios" ? 8 : 13,

```

Comment:
well, something must have changed, now it can be `top: 8` for both, in android is currently looking like this 
![image](https://user-images.githubusercontent.com/25725586/125052502-f16df500-e09b-11eb-8144-3185c47da0f1.png)